templates/deployment.yaml

```diff
@@ -1,5 +1,13 @@
+{{- if and (not (eq .Values.deploymentType "deployment")) (not (eq .Values.deploymentType "rollout")) }}
+{{- fail "Invalid deploymentType. Valid values are: deployment, rollout" }}
+{{- end }}
```

Comment:
Nit: I think these things make sense in a jsonschema or a helper if it's going to guard the entire template. Otherwise I would make this the last else on the kind: and api:  block.

Reply:
You're right! I think it's time to bring in jsonschema for this.